mesa/time.py

```diff
@@ -102,25 +112,31 @@ def get_agent_count(self) -> int:
         return len(self._agents)
 
     @property
-    def agents(self) -> list[Agent]:
-        return list(self._agents.values())
+    def agents(self) -> AgentSet:
+        # a bit dirty, but returns a copy of the internal agent set
+        return self._agents.select()
```

Comment:
This is indeed the way to do it for backwards compatibility. I don't like that `model.agents` is now an `AgentSet` and `schedule.agents` is a list (this PR doesn't change that). We should discuss how that align this in the long term.

Reply:
`AgentSet.select` returns an AgentSet, not a list. So, this PR already gives the desired alignment.

Because AgentSet is a Sequence, it should be backward compatible with places were lists are currently being used.

Reply:
Wait but then we have a behavior change, right? That makes this PR a little bit less backwards compatible I think.

Reply:
Not really, because AgentSet behaves like a list. You can get its length, index it, slice it, and iterate over it. That is why I made sure AgentSet implements all methods of Sequence.